manager/integration/tests/test_basic.py

```diff
@@ -2476,52 +2525,83 @@ def test_expansion_canceling(client, core_api, volume_name, pod):  # NOQA
-    write_pod_volume_data(core_api, pod_name, test_data)
+    test_data1 = generate_random_data(VOLUME_RWTEST_SIZE)
+    write_pod_volume_data(core_api, pod_name, test_data1, "test_file1")
 
     delete_and_wait_pod(core_api, pod_name)
     volume = wait_for_volume_detached(client, volume_name)
 
+    # Step 5 & 6: Try offline expansion, wait for the failure, then cancel it
     volume.expand(size=EXPAND_SIZE)
     wait_for_expansion_failure(client, volume_name)
     volume = client.by_id_volume(volume_name)
     volume.cancelExpansion()
     wait_for_volume_expansion(client, volume_name)
+    wait_for_volume_detached(client, volume_name)
     volume = client.by_id_volume(volume_name)
     assert volume.state == "detached"
     assert volume.size == SIZE
+    fix_replica_expansion_failure(client, core_api,
+                                  volume_name, EXPAND_SIZE, replicas)
 
-    # check if the volume still works fine
+    # Step 7: Verify the data after expansion cancellation
     create_and_wait_pod(core_api, pod)
-    resp = read_volume_data(core_api, pod_name)
-    assert resp == test_data
-    test_data = generate_random_data(VOLUME_RWTEST_SIZE)
-    write_pod_volume_data(core_api, pod_name, test_data)
+    resp = read_volume_data(core_api, pod_name, "test_file1")
+    assert resp == test_data1
+    test_data2 = generate_random_data(VOLUME_RWTEST_SIZE)
+    write_pod_volume_data(core_api, pod_name, test_data2, "test_file2")
 
-    # retry expansion
-    delete_and_wait_pod(core_api, pod_name)
-    volume = wait_for_volume_detached(client, volume_name)
+    # Step 8: Prepare to fail online expansion for one replica
+    volume = client.by_id_volume(volume_name)
+    replicas = volume.replicas
+    fail_replica_expansion(client, core_api,
+                           volume_name, EXPAND_SIZE, replicas)
+    # Step 9 & 10: Try online expansion, wait for the failure, then cancel it
     volume.expand(size=EXPAND_SIZE)
+    wait_for_expansion_failure(client, volume_name)
+    volume = client.by_id_volume(volume_name)
+    volume.cancelExpansion()
     wait_for_volume_expansion(client, volume_name)
     volume = client.by_id_volume(volume_name)
-    assert volume.state == "detached"
-    assert volume.size == str(EXPAND_SIZE)
-
-    create_and_wait_pod(core_api, pod)
+    assert volume.size == SIZE
+    fix_replica_expansion_failure(client, core_api,
+                                  volume_name, EXPAND_SIZE, replicas)
+
+    # Step 11: Validate the data content again
+    resp = read_volume_data(core_api, pod_name, "test_file1")
+    assert resp == test_data1
+    resp = read_volume_data(core_api, pod_name, "test_file2")
+    assert resp == test_data2
+
+    # Without a short wait, the volume replicas consider that the directory
+    # created in func fix_replica_expansion_failure still exists then fail
+    # the following expansion request.
+    # Not sure if it's caused by the caching.
+    time.sleep(5)
```

Comment:
Looks like an improvement in stability, should we create an issue for this?

Reply:
Not sure. I spent almost one day resolving this but no luck.

Reply:
The culprit of this wait: https://github.com/longhorn/longhorn/issues/5169#issuecomment-1367202468
Will fix the issue and remove this wait